### PR TITLE
Create new breadcrumb component

### PR DESCRIPTION
**Pull request by nathan-barrett**
## Description
Created a new breadcrumb component which will be initially used in theVPN Resource Center, but will most likely be used in other areas on mozilla.org. 

- [x] I have documented this change in the design system.
- [x] I have recorded this change in `CHANGELOG.md`.

### Testing

Start your local server and navigate to http://localhost:3000/patterns/molecules/breadcrumb.html# and compare to current Figma [design](https://www.figma.com/file/xo5JhiQIlpGWkb9lgh0ccZ/VPN-Resource-Guide?node-id=16%3A256)
Screen Shot 2021-09-29 at 5 16 55 PM
Short text on mobile sizing:
Screen Shot 2021-09-29 at 5 17 05 PM

### CODE REVIEW

**Comment by nathan-barrett** [comment]
There are still some small decisions / changes that need to be made before this would be ready to be merged. In addition, this is more of a first pass at a new protocol component and am happy to make changes to improve this.

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (craigcook) [inline src/assets/sass/protocol/components/_breadcrumb.scss:8]: This should be a token, and preferably a theme property.

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (craigcook) [inline src/assets/sass/protocol/components/_breadcrumb.scss:10]: Use tokens for padding.

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (craigcook) [inline src/assets/sass/protocol/components/_breadcrumb.scss:25]: Use tokens for margins, but our smallest spacing token is 4px. However, it might be better to use ems for this so it's proportional to the font size. Or maybe ch.

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (craigcook) [inline src/assets/sass/protocol/components/_breadcrumb.scss:23]: We probably shouldn't use the ">" character because it isn't actually an arrow, it already has a meaning (greater than). We can find some other unicode arrow character or maybe an SVG icon.

We also need to accommodate RTL and reverse the direction.

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (craigcook) [inline src/assets/sass/protocol/components/_breadcrumb.scss:33]: Tokens for spacing.

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (craigcook) [inline src/assets/sass/protocol/components/_breadcrumb.scss:20]: Support for `:not` is really recent, we should probably find a more backwards-compatible method.

**Review by craigcook — CHANGES_REQUESTED** [review]
I haven't done any real testing yet, just adding a bunch of drive-by comments on a first look through the code.

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (nathan-barrett) [reply src/assets/sass/protocol/components/_breadcrumb.scss:20]: https://developer.mozilla.org/en-US/docs/Web/CSS/:not seems like its the `selector list` that isn't well supported (`:not(div, .my-class)`) when its just one selector it seems like its supported back to IE 9. Should we still find something else to use?

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (nathan-barrett) [reply src/assets/sass/protocol/components/_breadcrumb.scss:23]: Ok, good call out

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (craigcook) [reply src/assets/sass/protocol/components/_breadcrumb.scss]: Ah, right you are. When I checked caniuse I was looking at the entry for multi-selector, but we should be fine with a single selector. My mistake 👍

File: src/patterns/molecules/breadcrumb/breadcrumb.hbs
Comment (maureenlholland) [reply src/patterns/molecules/breadcrumb/breadcrumb.hbs:3]: "Current implementation is to use this component as a header" - I'm not following here, does this mean it should be at the top of the page as opposed to the bottom? Or it should be part of the header tag?

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (maureenlholland) [inline src/assets/sass/protocol/components/_breadcrumb.scss:13]: I think we're overriding the bold font type in the `-item`, `-title` selector below with 400 font weight

Comment (maureenlholland) [reply src/assets/sass/protocol/components/_breadcrumb.scss:13]: I wonder too if it's worthwhile having separate classes for item and title or if we could use `:first-child` and `.mzp-c-breadcrumb-item + .mzp-c-breadcrumb-item` selectors to target the first and subsequent items in a way that could scale to multiple items in the breadcrumb.

**Review by maureenlholland — CHANGES_REQUESTED** [review]
Great start! Only one actual requested changes for an overridden font-weight style and typo. The rest are just musings to take or leave.

Also, something we might want to ping Elise on. What to do on mobile when the breadcrumb is longer than one line? I don't know if we want a default wrap because it takes up such valuable space. Baymard seems to suggest a horizontal scroll is acceptable but I don't know if that's better. (Baymard doesn't link titles, but it's the How to Implement Long Breadcrumb Paths section: https://baymard.com/blog/implementing-mobile-hierarchy-breadcrumbs)
breadcrumb-mobile

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (maureenlholland) [inline src/assets/sass/protocol/components/_breadcrumb.scss:33]: nit: remove extra empty lines

File: src/patterns/molecules/breadcrumb/breadcrumb.hbs
Comment (maureenlholland) [inline src/patterns/molecules/breadcrumb/breadcrumb.hbs:3]:
```suggestion
description: A type of navigation menu used to provide a list of links based on current page within the site hierarchy, allowing users to visit previous pages from the current page.
```

**Review by maureenlholland — CHANGES_REQUESTED** [review]
tiny nitpicks, and will need a changelog update - but more changelog conflict is coming with https://github.com/mozilla/protocol/pull/731

@craigcook do you think this or the protocol-assets dep update PR will merge first?

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (craigcook) [inline src/assets/sass/protocol/components/_breadcrumb.scss:25]: Nit: omit leading zeroes on decimals (doesn't matter in the grand scheme but we should be consistent).

**Review by craigcook — CHANGES_REQUESTED** [review]

**Comment by nathan-barrett** [comment]
This will probably need another rebase after 15.1 goes through, but I've double checked with Elise on the comments brought up in this PR and I _think_ have everything resolved. Let me know if anything else stands out. Would love to get this merged so we can use this when the VPN resource center goes live at end of Nov.

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (maureenlholland) [inline src/assets/sass/protocol/components/_breadcrumb.scss:34]: Not sure we want `-alt` here as it's pretty light: https://github.com/mozilla/protocol/blob/main/src/assets/sass/protocol/includes/_themes.scss#L56

Screen Shot 2021-11-03 at 11 01 10 AM

Unless we also set color for `body-text-color-alt`? Or maybe change background to use `background-color-alt-inverse`?

**Review by maureenlholland — CHANGES_REQUESTED** [review]
Thanks for checking in with Elise and following up on this! Would love to see it go into the wild with VRC 🍞 

A couple of nitty things and a question about dark theme colors. This is looking pretty good to me though.

File: src/assets/sass/protocol/components/_breadcrumb.scss
Comment (maureenlholland) [reply src/assets/sass/protocol/components/_breadcrumb.scss:34]: Oh, yep, just saw a comment on this in a previous review. Prefer `background-color-alt-inverse`

**Comment by craigcook** [comment]
I just came across this: https://html.spec.whatwg.org/multipage/semantics-other.html#rel-up which sent me down a short rabbit-hole, and I quickly found https://css-tricks.com/markup-for-breadcrumbs/ and then https://schema.org/BreadcrumbList

So in short, it seems like a paragraph with links may be preferable to a list. An ordered list does at least clearly indicate a sequence, but it doens't indicate hierarchy: the links are in a specific order but there's no parent<>child relationship. 

Just putting links in order without marking them up as a list already establishes the sequence without the added implication that they're all at the same level. Probably the best way to establish a hierarchy of links would be a series of nested lists, but that seems really clunky and could create more problems than it solves.

So then there's the idea of adding some more attributes to better describe the relationship. There was a proposed `rel="up"` attribute that seemed promising until it was axed. There's a breadcrumb schema at Schema.org but it's also clunky and seems like it helps search engines more than it helps humans.

So I'm a little torn on whether we should stick with an ordered list or just do a paragraph with inline links, and I could go either way.

But this little bit of research *has* convinced me that we should require breadcrumbs to be in a `nav` after all, since they are absolutely navigation and always will be. So I suggest `<nav class="mzp-c-breadcrumb">` as the component container.

File: src/patterns/molecules/breadcrumb/breadcrumb.hbs
Comment (craigcook) [inline src/patterns/molecules/breadcrumb/breadcrumb.hbs:10]: I also [commented elsewhere](https://github.com/mozilla/protocol/pull/730#issuecomment-961258254) but repeating here so it's not lost: after a little research I think we should indeed require breadcrumbs to be in a `nav` and make this `<nav class="mzp-c-breadcrumb" aria-label="breadcrumb">`

File: src/patterns/molecules/breadcrumb/breadcrumb.hbs
Comment (craigcook) [inline src/patterns/molecules/breadcrumb/breadcrumb.hbs:11]: With the main component class moving to the `nav` I think this would become `mzp-c-breadcrumb-list`

File: src/patterns/molecules/breadcrumb/breadcrumb.hbs
Comment (craigcook) [inline src/patterns/molecules/breadcrumb/breadcrumb.hbs:13]: I wonder if we really need a class on each link. Maybe any link in an `mzp-c-breadcrumb-item` gives us enough description and specificity?

**Review by craigcook — CHANGES_REQUESTED** [review]

**Comment by maureenlholland** [comment]
I'd stick with an ordered list for now, that seems to be the accepted accessible pattern: https://github.com/scottaohara/a11y_breadcrumbs

But fascinating the HTML spec has it in paragraphs. I'm not 100% opposed to the idea, but I think the ol does provide a bit of a sense of hierarchy (like the first item is the 1st level page and the 2nd item is 2nd level)

**Comment by nathan-barrett** [comment]
Oh, nice - thanks for the links @craigcook - I'll take a look and see which direction makes the most sense. Maybe an `ol` rather than a `ul` at the very least.

File: src/patterns/molecules/breadcrumb/breadcrumb.hbs
Comment (nathan-barrett) [reply src/patterns/molecules/breadcrumb/breadcrumb.hbs:13]: Yeah I think you're right.

**Comment by craigcook** [comment]
Yeah, an ordered list feels right to me. Let's stick with that.

**Review by maureenlholland — APPROVED** [review]
Pending rebase and approval from @craigcook, but everything related to breadcrumb looks good to me! r+ 🏆

**Review by craigcook — APPROVED** [review]
LGTM! 🍞

**Pull request merged**
